lib/al/Library/Area/AreaShapeCube.cpp

```diff
@@ -0,0 +1,36 @@
+#include "al/Library/Area/AreaShapeCube.h"
+#include <math/seadBoundBox.h>
+
+namespace al {
+
+AreaShapeCube::AreaShapeCube(al::AreaShapeCube::OriginType originType) : mOriginType(originType) {}
+
+bool AreaShapeCube::isInVolume(const sead::Vector3f& trans) const {
+    sead::Vector3f localPos = sead::Vector3f::zero;
+    calcLocalPos(&localPos, trans);
+
+    return isInLocalVolume(localPos);
+}
+
+// bool AreaShapeCube::isInVolumeOffset(const sead::Vector3f& trans, f32 offset) const {}
+// bool AreaShapeCube::calcNearestEdgePoint(sead::Vector3f* edgePoint, const sead::Vector3f& trans) const {}
+// bool AreaShapeCube::checkArrowCollision(sead::Vector3f*, sead::Vector3f*, const sead::Vector3f&, const sead::Vector3f&) const {}
+// bool AreaShapeCube::calcLocalBoundingBox(sead::BoundBox3f* boundingBox) const {}
+
+bool AreaShapeCube::isInLocalVolume(const sead::Vector3f& trans) const {
+    float bottom = mOriginType == OriginType::Base ? 0.0f : (mOriginType == OriginType::Top ? -1000.0f : 500.0f);
+    float top = mOriginType == OriginType::Base ? 1000.0f : (mOriginType == OriginType::Top ? 0.0f : 500.0f);
```

Comment:
Doing a `switch` might be easier to look at, might even combine `bottom` and `top` in a single `switch`.

Reply:
`if`/`else if`/`else` works here, i can't it to match when combining them into one but it works when they're separate. `switch` works too but i think `if`-`else` looks better personally. i'll combine it with the suggestion below on decomp.me